packages/std/src/math/decimal.rs

```diff
@@ -1134,6 +1151,23 @@ mod tests {
         );
     }
 
+    #[test]
+    fn decimal_iter_sum() {
+        let items = vec![
+            Decimal::zero(),
+            Decimal(Uint128::from(2u128)),
+            Decimal(Uint128::from(2u128)),
+        ];
+        assert_eq!(items.iter().sum::<Decimal>(), Decimal(Uint128::from(4u128)));
```

Comment:
Is the turbofish `::<Decimal>` necessary here? Rust can't infer the type because you could possibly want to compare `Decimal` with something else?

Reply:
Checked it again. Seems necessary:

```
error[E0283]: type annotations needed
    --> packages/std/src/math/decimal.rs:1161:33
     |
1161 |         assert_eq!(items.iter().sum(), Decimal(Uint128::from(4u128)));
     |                                 ^^^ cannot infer type for type parameter `S` declared on the associated function `sum`
     |
     = note: cannot satisfy `_: std::iter::Sum<&decimal::Decimal>`
help: consider specifying the type argument in the method call
     |
1161 |         assert_eq!(items.iter().sum::<S>(), Decimal(Uint128::from(4u128)));
     |                                    +++++

error[E0283]: type annotations needed
    --> packages/std/src/math/decimal256.rs:1245:26
     |
1245 |             items.iter().sum(),
     |                          ^^^ cannot infer type for type parameter `S` declared on the associated function `sum`
     |
     = note: cannot satisfy `_: std::iter::Sum<&decimal256::Decimal256>`
help: consider specifying the type argument in the method call
     |
1245 |             items.iter().sum::<S>(),
     |                             +++++

For more information about this error, try `rustc --explain E0283`.
```